shop/src/main/java/ro/msg/learning/shop/utils/mapper/OrderMapper.java

```diff
@@ -0,0 +1,23 @@
+package ro.msg.learning.shop.utils.mapper;
+
+import ro.msg.learning.shop.dto.OrderDto;
+import ro.msg.learning.shop.model.Customer;
+import ro.msg.learning.shop.model.Location;
+import ro.msg.learning.shop.model.Order;
+
+import java.time.LocalDate;
+
+public class OrderMapper {
+    public static Order DtoToOrder(OrderDto orderDto) {
+        if (orderDto == null)
+            return null;
+        Location location = new Location(orderDto.getName(), orderDto.getCountry(), orderDto.getCity(),
+                orderDto.getCounty(), orderDto.getStreetAddress());
+        location.setId(orderDto.getLocationId());
+        Customer customer = new Customer(orderDto.getCustomer().getFirstName(), orderDto.getCustomer().getLastName(),
```

Comment:
First of all why would you create a new customer by using all the details of the customer within orderDto? If there's a reason behind it you could change the constructor to receive only one parameter (`orderDto.getCustomer()`) and do the value retrievals in there